java11-backend/src/main/scala/sttp/client/httpclient/HttpClientBackend.scala

```diff
@@ -0,0 +1,186 @@
+  InputStreamBody,
+  MultipartBody,
+  NoBody,
+  Request,
+  Response,
+  ResponseAs,
+  ResponseAsByteArray,
+  ResponseAsFile,
+  ResponseAsStream,
+  ResponseMetadata,
+  StreamBody,
+  StringBody,
+  SttpBackend,
+  SttpBackendOptions
+}
+import sttp.model._
+
+import scala.collection.JavaConverters._
+import scala.language.higherKinds
+import scala.util.{Failure, Try}
+
+abstract class HttpClientBackend[F[_], S](client: HttpClient) extends SttpBackend[F, S, WebSocketHandler] {
+
+  private[httpclient] def convertRequest[T](request: Request[T, S]) = {
+    val builder = HttpRequest
+      .newBuilder()
+      .uri(request.uri.toJavaUri)
+
+    builder.method(request.method.method, bodyToHttpBody(request, builder))
+    request.headers
+      .filterNot(_.name == HeaderNames.ContentLength)
+      .foreach(h => builder.header(h.name, h.value))
+    builder.timeout(JDuration.ofMillis(request.options.readTimeout.toMillis)).build() //TODO is this correct?
+  }
+
+  private def bodyToHttpBody[T](request: Request[T, S], builder: HttpRequest.Builder) = {
+    request.body match {
+      case NoBody                => BodyPublishers.noBody()
+      case StringBody(b, _, _)   => BodyPublishers.ofString(b)
+      case ByteArrayBody(b, _)   => BodyPublishers.ofByteArray(b)
+      case ByteBufferBody(b, _)  => BodyPublishers.ofByteArray(b.array())
+      case InputStreamBody(b, _) => BodyPublishers.ofInputStream(() => b)
+      case FileBody(f, _)        => BodyPublishers.ofFile(f.toFile.toPath)
+      case StreamBody(s)         => streamToRequestBody(s)
+      case MultipartBody(parts) =>
+        val multipartBodyPublisher = multipartBody(parts)
+        builder.header(HeaderNames.ContentType, s"multipart/form-data; boundary=${multipartBodyPublisher.getBoundary}")
+        multipartBodyPublisher.build()
+    }
+  }
+
+  def streamToRequestBody(stream: S): BodyPublisher = throw new IllegalStateException("Streaming isn't supported")
+
+  private def multipartBody[T](parts: Seq[Part[BasicRequestBody]]) = {
+    val multipartBuilder = new MultiPartBodyPublisher()
+    parts.foreach { p =>
+      val allHeaders = p.headers :+ Header(HeaderNames.ContentDisposition, p.contentDispositionHeaderValue)
+      p.body match {
+        case FileBody(f, _) =>
+          multipartBuilder.addPart(p.name, f.toFile.toPath, allHeaders.map(h => h.name -> h.value).toMap.asJava)
+        case StringBody(b, _, _) =>
+          multipartBuilder.addPart(p.name, b, allHeaders.map(h => h.name -> h.value).toMap.asJava)
+      }
+    }
+    multipartBuilder
+  }
+
+  private[httpclient] def readResponse[T](
+      res: HttpResponse[Array[Byte]],
+      responseAs: ResponseAs[T, S]
+  ): F[Response[T]] = {
+    val headers = res
+      .headers()
+      .map()
+      .keySet()
+      .asScala
+      .flatMap(name => res.headers().map().asScala(name).asScala.map(Header(name, _)))
+      .toList
+
+    val code = StatusCode(res.statusCode())
+    val message = "???" // TODO
```

Comment:
todo :)

Reply:
Yes :) I don't know how to extract this message from the response